### DISCUSSION TITLE
How to get the channel Id and Access hash

### DISCUSSION THREAD

**Discussion by jiyamathias** [gotd/td#997] [Q&A]
I want to use the method   `api.ChannelsUpdateUsername()` but it requests payload is 
```go
type ChannelsUpdateUsernameRequest struct {
	// Channel
	Channel InputChannelClass
	// New username
	Username string
}
``` 
and this is the InputChannelClass object below
``` go
type InputChannel struct {
	// Channel ID
	ChannelID int64
	// Access hash taken from the channel¹ constructor
	//
	// Links:
	//  1) https://core.telegram.org/constructor/channel
	AccessHash int64
}
```
I don't know how to get the ChannelID and AccessHash for the InputChannel object.
I have been stuck here for days now and don't know how to move forward.

**Comment by zthreee** [comment]
This is how I hanlded it - not sure if there is a better way, but works for now: 

```
func getChannelName(ctx context.Context, client *tg.Client, channelID int64) (string, int64, error) {
	inputChannel := &tg.InputChannel{
		ChannelID:  channelID,
		AccessHash: 0,
	}
	channels, err := client.ChannelsGetChannels(ctx, []tg.InputChannelClass{inputChannel})

	if err != nil {
		return "", 0, fmt.Errorf("failed to fetch channel: %w", err)
	}

	if len(channels.GetChats()) == 0 {
		return "", 0, fmt.Errorf("no channels found")
	}

	channel := channels.GetChats()[0].(*tg.Channel)
	return channel.Username, channel.AccessHash, nil
}
```

**Reply by idkroff** [reply]
You're genius.

**Reply by RoninInPC** [reply]
This don't working. A HASH OF ZERO IS RETURNED. Most likely, something else needs to be done.

**Reply by RoninInPC** [reply]
Problem find, delete -100 in prefix channel

**Reply by ip75** [reply]
Super! Thank you very much bro!

**Comment by jiyamathias** [comment]
Thanks for taking your time to respond to me. 
Much appreciation 🙏

Although I have a question, how do I get the channel ID?

**Reply by RoninInPC** [reply]
One of the ways to open telegram web and see the url of the channel, there will be a number.

**Reply by ip75** [reply]
```bash
curl https://api.telegram.org/bot<botAPI_hash>/getUpdates | jq
```
looking for chat section in response:
```json
        "chat": {
          "id": -100XXXXXXXXXXX,
          "title": "<channel name>",
          "is_forum": true,
          "type": "supergroup"
        },

```
for mtproto chatID is a chat.id without -100 so  
```go
mtprotoChatID = -chat.id-1000000000000
```
